sdk/azcore/runtime/pager.go

```diff
@@ -88,3 +90,25 @@ func (p *Pager[T]) NextPage(ctx context.Context) (T, error) {
 func (p *Pager[T]) UnmarshalJSON(data []byte) error {
 	return json.Unmarshal(data, &p.current)
 }
+
+// FetcherForNextLink is a helper containing boilerplate code to simplify creating a PagingHandler[T].Fetcher from a next link URL.
+func FetcherForNextLink(ctx context.Context, pl Pipeline, nextLink string, createReq func(context.Context) (*policy.Request, error)) (*http.Response, error) {
+	var req *policy.Request
+	var err error
+	if nextLink == "" {
+		req, err = createReq(ctx)
+	} else if nextLink, err = EncodeQueryParams(nextLink); err == nil {
```

Comment:
This all looks good to me.
One thing that is supposed to happen and I can't tell if it is with this code is that the api-version query parameter value should be changed to the same value used by the client and not remain with the same value as was in the nextLink URL.
The scenario is that a client using api-version=v1 initiates the paging and then a client using api-version=v2 gets the next page.
This is also how LROs should work. The Poller does a GET to a URL and the api-version value should reflect that of the current client used when creating the poller.

Reply:
We don't do that at present.  I've opened https://github.com/Azure/azure-sdk-for-go/issues/21408 to track that.

Reply:
ok